messaging/src/main/java/org/axonframework/messaging/unitofwork/LegacyMessageSupportingContext.java

```diff
@@ -26,27 +27,31 @@
 import java.util.function.UnaryOperator;
 
 /**
- * No-op implementation of the {@link ProcessingContext}.
+ * Legacy implementation of the {@link ProcessingContext} that should only be used for legacy components. Can only be
+ * constructed with a {@link org.axonframework.messaging.Message} as parameter that serves as his only resource. Any
+ * operation except getting the message from the context will fail.
  *
- * @author Allard Buijze
+ * @author Mitchell Herrijgers
  * @since 5.0.0
+ * @deprecated Only in use for legacy "sync" components.
  */
-public class NoProcessingContext implements ProcessingContext {
-
-    /**
-     * Constant of the {@link NoProcessingContext} to be used as a single reference.
-     */
-    public static final NoProcessingContext INSTANCE = new NoProcessingContext();
+@Deprecated
+public class LegacyMessageSupportingContext implements ProcessingContext {
```

Comment:
Just want to confirm the following statement with you, so that I am certain I am on the right track:

- The `LegacyMessageSupportingContext` should be used by **everybody** that uses the infrastructure components we move to legacy/migration.
- The `LegacyMessageSupportingContext` should **never** be used with new components.

Is that a correct assumption from my part?

Reply:
Yes, it's more or less a 'polyfill'. This is used in web components, where we want to use browser functions that are not available in older versions, but can be emulated. Usually at a small (performance) cost. 
This is similar. That's why I immediately deprecated it.

Reply:
Gotcha, thanks for that :-)
I anticipate to see this short piece of explanation pop up in the `api-changes.md` somewhere! ^_^